components/apps_form/apps_form_component.tsx

```diff
@@ -307,6 +311,7 @@ export class AppsForm extends React.PureComponent<Props, State> {
         const values = {...this.state.values, [name]: value};
 
         if (field.refresh) {
+            this.setState({changing: true});
             this.props.actions.refreshOnSelect(field, values).then((res) => {
                 if (res.error) {
```

Comment:
Is an error happens, we'll want to restore the visibility of the form

Reply:
Any reason why we don't just unconditionally set `changing` to `false` at the beginning of this callback?

Reply:
That's makes better sense.  Done.

Reply:
Can we change `changing` to `refreshing`?

Reply:
How about loading?

Reply:
I'll spin up an instance so we can all see the current state.

Reply:
If the only time we are "loading" is when we are refreshing, then 1/5 we should use `refreshing` to be clear about what's happening.